tests/test_bind_edge_cases.py

```
@@ -336,3 +336,243 @@ def foo(a: int, b: int, c: int) -> int:
+
+        @node(output_name="result")
+        def typed_func(x: int, y: str) -> str:
+            return f"{x}: {y}"
+
+        g = Graph([typed_func])
+        g_bound = g.bind(y="hello")
+        g_unbound = g_bound.unbind("y")
+
+        # Type info should still be available in the underlying node
+        typed_node = g_unbound.nodes["typed_func"]
+        assert typed_node.parameter_annotations.get("y") == str
```

Ruff (0.14.13)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)